eng/scripts/SdkVersion-Common.ps1

```diff
@@ -7,4 +7,4 @@ $VersionStringRegEx = '(#define AZ_SDK_VERSION_STRING )"(([0-9]+)\.([0-9]+)\.([0
 $VersionMajorRegEx = '(#define AZ_SDK_VERSION_MAJOR )([0-9]+)';
 $VersionMinorRegEx = '(#define AZ_SDK_VERSION_MINOR )([0-9]+)';
 $VersionPatchRegEx = '(#define AZ_SDK_VERSION_PATCH )([0-9]+)';
-$VersionPrereleaseRegEx = '(#(un)?def(ine)? AZ_SDK_VERSION_PRERELEASE) ?("[a-z0-9.]+")?';
+$VersionPrereleaseRegEx = '(#define AZ_SDK_VERSION_PRERELEASE ?("[a-z0-9.]*")?)((\r\n|\r|\n)#undef AZ_SDK_VERSION_PRERELEASE)?';
```

Comment:
How do we test this PR to see that it does the right thing?

Reply:
Test results in practice: https://github.com/Azure/azure-sdk-for-c/pull/2304#issuecomment-1211410474